marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ProvenanceController.java

```diff
@@ -0,0 +1,70 @@
+package com.marklogic.hub.central.controllers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.dataservices.ModelsService;
+import com.marklogic.hub.dataservices.ProvenanceService;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.Date;
+import java.util.List;
+
+@Controller
+@RequestMapping("/api/provenance")
+public class ProvenanceController extends BaseController {
+    /**
+     * Returns the provenance graph for a document.
+     *
+     * @return
+     */
+    @RequestMapping(value = "/getProvenanceGraph", method = RequestMethod.GET)
+    @ResponseBody
+    @ApiOperation(value = "Get the provenance data for a document in graph form", response = ProvenanceGraph.class)
+    public ResponseEntity<JsonNode> getProvenanceGraph(@RequestParam String documentURI) {
```

Comment:
Any reason why we are not pssing it as PathVariable? In most endpoints I saw we usually pass the "id"/"name" as part of the path.
I don't know the characters a documentURI can have. But to include dot you can do /{documentURI:.+}
Reference: https://www.baeldung.com/spring-mvc-pathvariable-dot

Reply:
The documentURI can be anything that complies with the URI standard which includes protocols, forward slashes, etc. I think it makes more sense, in this case, to have it be a request parameter to avoid potential issues with encoding and parsing the value.